docs/src/further_topics/netcdf_io.rst

```diff
@@ -122,6 +122,61 @@ Iris' optimisation all together, and will take its chunksizes from Dask's behavi
     (70, 37, 49)
 
 
+Variable-length datatypes
+-------------------------
+
+The NetCDF4 module provides support for variable-length (or "ragged") data
+types (``VLType``); see
+`Variable-length data types <https://unidata.github.io/netcdf4-python/#variable-length-vlen-data-types>`_
+
+The ``VLType`` allows for storing data where the length of the data in each array element
+can vary. When ``VLType`` arrays are loaded into Iris cubes (or numpy), they are stored
+as an array of ``Object`` types - essentially an array-of-arrays, rather than a single
+multi-dimensional array.
+
+The most likely case to encounter variable-length data types is when an array of
+strings (not characters) are stored in a NetCDF file. As the string length for any
+particular array element can vary the values are stored as an array of ``VLType``.
+
+As each element of a variable-length array is stored as a ``VLType`` containing
+an unknown number of vales, the total size of a variable-length NetCDF array
+cannot be known without first loading the data. This makes it difficult for
+Iris to make an informed decision on whether to the load the data lazily or not.
+The user can aid this decision using *VLType size hinting* described below.
+
+VLType size hinting
+^^^^^^^^^^^^^^^^^^^
+
+If the user has some *a priori* knowledge of the average length of the data in
+variable-length ``VLType``, this can be provided as a hint to Iris via the
+``CHUNK_CONTROL`` context manager and the special ``_vl_hint`` keyword
+targeting the variable, e.g. ``CHUNK_CONTROL.set("varname", _vl_hint=5)``.
+This allows Iris to make a more informed decision on whether to load the
+data lazily.
```

Comment:
This section is great now, but I still have a slight doubt about this last bit.

In my mind, the primary purpose is to decide chunk sizes, not whether to make the content lazy at all -- that's just a bonus in special cases.

How do you feel about tweaking this ?
If you disagree, or you don't have time now, we can be pragmatic + merge as-is !

Reply:
The `_vl_hint` only affects the lazy loading calculation so I think this comment is still pertinent.

Reply:
OK I get it now.

I had been working under the mistaken assumption that we somehow passed an estimated 'itemsize' to the Iris automatic chunking algorithm in "as_lazy_data".
But now I understand that this only uses "var.shape" and "dtype.itemsize", so it basically makes a nonsense of the variable-length cases.
Potentially we could do something better by passing the "_vl_hint" into that calculation too, but perhaps that is a step too far ?
What the user **_can_** (already) do is to take control of the chunking for the variable, but it must be complete control since otherwise limiting chunking in one dimension will cause Iris to expand the chunksize in other dims.